src/compiler/emitter.ts

```diff
@@ -168,155 +167,60 @@ module ts {
-                    if (char !== CharacterCodes.i && char !== CharacterCodes.n) {
-                        if (tempCount < 26) {
-                            tempName = "_" + String.fromCharCode(char);
-                        }
-                        else {
-                            tempName = "_" + (tempCount - 26);
-                        }
+                while (true) {
+                    let count = tempFlags & TempFlags.CountMask;
+                    let ch = CharacterCodes.a + count;
+                    let name = count < 26 ? "_" + String.fromCharCode(ch) : "_" + (count - 26);
+                    tempFlags++;
+                    if (ch !== CharacterCodes.i && ch !== CharacterCodes.n && isUniqueName(name)) {
+                        return name;
                     }
-
-                    tempCount++;
                 }
-                while (resolver.resolvesToSomeValue(location, tempName));
-
-                return tempName;
             }
 
-            // Generates a name that is unique within current file and does not collide with
-            // any names in global scope.
-            // NOTE: names generated by 'makeTempVariableName' and 'makeUniqueName' will never conflict
-            // because of the way how these names are generated
-            // - makeUniqueName builds a name by picking a base name (which should not be empty string)
-            //   and appending suffix '_<number>'
-            // - makeTempVariableName creates a name using the following pattern:
-            //   _a .. _h, _j ... _z, _0, _1, ...
-            // This means that names from 'makeTempVariableName' will have only one underscore at the beginning
-            // and names from 'makeUniqieName' will have at least one underscore in the middle 
-            // so they will never collide.
+            // Generate a name that is unique within the current file and doesn't conflict with any names
+            // in global scope. The name is formed by adding an '_n' suffix to the specified base name,
+            // where n is a positive integer. Note that names generated by makeTempVariableName and
+            // makeUniqueName are guaranteed to never conflict.
             function makeUniqueName(baseName: string): string {
-                Debug.assert(!!baseName);
-
                 // Find the first unique 'name_n', where n is a positive number
                 if (baseName.charCodeAt(baseName.length - 1) !== CharacterCodes._) {
                     baseName += "_";
                 }
-
                 let i = 1;
-                let generatedName: string;
                 while (true) {
-                    generatedName = baseName + i;
+                    let generatedName = baseName + i;
                     if (isUniqueName(generatedName)) {
-                        break;
+                        return (generatedNameSet || (generatedNameSet = {}))[generatedName] = generatedName;

```

Comment:
This seems overly dense.  It took me several reads just to figure out what it was doing.  Can we simplify is as something more like

```ts
let set = generatedNameSet || (generatedNameSet = {});
set[generatedName] = generatedName;
```
This helps break out the different concerns.  I.e. lazily initializing the set, and adding the element to the set.